app/src/main/java/com/example/ecodigify/ui/search/SearchFragment.kt

```diff
@@ -67,71 +72,96 @@ class SearchFragment : Fragment() {
             }
 
             override fun onQueryTextChange(newText: String?): Boolean {
-
                 if (newText != null) {
-                    // TODO: actually update the shown recipes from manager using filtered ingredients
-                    updateRecipes(emptyArray())
+                    searchJob?.cancel()
+                    searchJob = run(
+                        lifecycle = lifecycle,
+                        function = {
+                            delay(250) // Debounce
+                            search = newText
+                            Manager.find(search)
+                        },
+                        done = { recipes -> updateRecipes(recipes.toTypedArray()) },
+                        error = { updateRecipes(emptyArray()) }
+                    )
                 }
+
                 return true
             }
         })
 
-
-        // TODO: get some ingredients from the manager?
-        val ingredients: Array<Ingredient> = arrayOf(
-            Ingredient(1, "ing XD", LocalDate.now(), LocalDate.now(), emptyList(), "4"),
-            Ingredient(2, "ing D:", LocalDate.now(), LocalDate.now(), emptyList(), "5"),
-            Ingredient(3, "ing :D", LocalDate.now(), LocalDate.now(), emptyList(), "6"),
-        )
-
         val popFilterMenu = PopupMenu(activity, filterButton)
 
-        for (ing in ingredients) {
-            val itm = popFilterMenu.menu.add(Menu.NONE, ing.id.toInt(), Menu.NONE, ing.name) // Cast might be problematic!
-            itm.isCheckable = true
-            itm.isChecked = true
-        }
+        run(
+            lifecycle = lifecycle,
+            function = Manager::ingredientGetAll,
+            done = { ingredients ->
+
+                for (ing in ingredients) {
+                    val itm = popFilterMenu.menu.add(
+                        Menu.NONE,
+                        ing.id.toInt(),
+                        Menu.NONE,
+                        ing.name
+                    ) // Cast might be problematic!
+                    itm.isCheckable = true
+                    itm.isChecked = true
+                }
```

Comment:
```suggestion
                for (i in ingredients.indices) {
                    val itm = popFilterMenu.menu.add(
                        Menu.NONE,
                        i,
                        Menu.NONE,
                        ingredients[i].name
                    )
                    itm.isCheckable = true
                    itm.isChecked = true
                }
```
Questo potrebbe fixare il cast, in caso possiamo aggiungere che `ingredientGetAll` li ritorni ordinati